src/js/arrays_loops.js

```diff
@@ -0,0 +1,131 @@
+import { contacts } from "../data/contacts.js";
+
+console.log(contacts);
+
+// Array Order Methods
+
+// 1.	Write a function that takes the array of contacts and returns a new array sorted by last name.
+
+const contactsArray = contacts.results;
+
+function sortedByLastName(contactsArray) {
+  return contactsArray.sort((a, b) => {
+    const lastnameA = a.name.last.toLowerCase();
+    const lastnameB = b.name.last.toLowerCase();
+
+    if (lastnameA > lastnameB) {
+      return 1;
+    }
+    if (lastnameA < lastnameB) {
+      return -1;
+    }
+    return 0;
+  });
+}
+
+console.log(sortedByLastName(contactsArray));
+
+// 2.	Write a function to return the array of contacts in reverse order. Do not use the built-in reverse() method.
+function reverseOrderArray(contactsArray) {
+  let contactObject = {};
+  let reversedOrderArray = [];
+  for (let i = contactsArray.length - 1; i >= 0; i--) {
+    contactObject = contactsArray[i];
+    reversedOrderArray.push(contactObject);
+  }
+  return reversedOrderArray;
+}
+
+console.log(reverseOrderArray(contactsArray));
+
+// 3.	Write a function that returns the first 5 contacts from the sorted list (by last name).
+function returnFirstFiveSortedContacts(contactsArray) {
+  const sortedByLastNameArray = sortedByLastName(contactsArray);
+  return sortedByLastNameArray.slice(0, 5);
+}
+
+console.log(returnFirstFiveSortedContacts(contactsArray));
+
+// 4.	Create a function that returns an array of all unique first names. No duplicates should be present.
+function returnAllUniqueFirstNames(contactsArray) {
+  let firstNamesArray = [];
```

Comment:
complexity of solution is too great.

please come up with more efficient solution.